opentracing-api/src/main/java/io/opentracing/Finishable.java

```diff
@@ -0,0 +1,32 @@
+package io.opentracing;
+
+/**
+ * {@link Finishable} factors out the overloaded {@link #finish()} method(s).
+ *
+ * @see ActiveSpan.Observer#onDeactivate(ActiveSpan, Finishable)
+ * @see Span
+ */
+public interface Finishable {
+    /**
+     * Sets the end timestamp to now and records the span.
+     *
+     * <p>With the exception of calls to {@link BaseSpan#context}, this should be the last call made to the span
+     * instance. Future calls to {@link #finish} are defined as noops, and future calls to methods other than
+     * {@link BaseSpan#context} lead to undefined behavior.
+     *
+     * @see BaseSpan#context()
+     */
+    void finish();
+
+    /**
+     * Sets an explicit end timestamp and records the span.
+     *
+     * <p>With the exception of calls to Span.context(), this should be the last call made to the span instance, and to
+     * do otherwise leads to undefined behavior.
+     *
+     * @param finishMicros an explicit finish time, in microseconds since the epoch
+     *
+     * @see BaseSpan#context()
+     */
+    void finish(long finishMicros);
```

Comment:
Would this be a good time to consider separating the units from the value?  Perhaps a signature like this:
`void finish(long finishTime, TimeUnit units)`
I feel having implied unit magnitude often results in confusion.  In the past, I've had to write code that dynamically figures out the proper magnitude assuming an incorrectly reported scale (seconds vs milliseconds) all because the api was used incorrectly.